src/strategies/razor-network-voting/index.ts

```diff
@@ -0,0 +1,125 @@
+  stake: BigNumber,
+  totalSupply: BigNumber,
+  amount: BigNumber
+) {
+  try {
+    return stake.mul(amount).div(totalSupply);
+  } catch (err) {
+    return BigNumber.from(0)
+    // do nothing
+  }
+}
+
+function wei_to_ether(amount: number) {
+  return amount / 10 ** 18;
+}
+
+export async function strategy(
+  space: any,
+  network: any,
+  provider: any,
+  addresses: any[],
+  options: any,
+  //symbol: string,
+  snapshot: string
+) {
+  const params = {
+    delegators: {
+      __args: {
+        where: {
+          delegatorAddress_in: addresses
+        } // delegatorAddress
+      }, // Amount_Delegated
+      staker: {
+        totalSupply: true,
+        stake: true
+      },
+      delegatorAddress: true,
+      sAmount: true
+    },
+    stakers: {
+      __args: {
+        where: {
+          staker_in: addresses //  stakerAddress
+        }
+      },
+      stake: true,
+      totalSupply: true,
+      staker: true,
+      sAmount: true
+    }
+  };
+
+  if (snapshot !== 'latest') {
+    // @ts-ignore
+    params.delegators.__args.block = { number: snapshot };
+  }
+
+  const score = {};
+
+  // subgraph request 1 : it fetches all the details of the stakers and delegators.
+  const result = await subgraphRequest(RAZOR_NETWORK_SUBGRAPH_URL, params);
+  console.log(result);
+  if (result) {
+    result.delegators.forEach(
+      async (delegator: {
+        sAmount: string;
+        staker: { stake: string; totalSupply: string };
+        delegatorAddress: string | number;
+      }) => {
+        const razor_amount = sRZR_to_RZR(
+          BigNumber.from(delegator.sAmount),
+          BigNumber.from(delegator.staker.totalSupply),
+          BigNumber.from(delegator.staker.stake)
+        );
+        //if delegator has delegated to more than one staker, we need to add that amount also to calculate score.
+        if (!score[delegator.delegatorAddress]) {
+          //if score[delegator] has no score setup already we will put it as intial amount
+          score[delegator.delegatorAddress] = wei_to_ether(Number(razor_amount));
+        } else {
+          // update the score of delegator by adding new Stoken -> razor Value
+          score[delegator.delegatorAddress] += wei_to_ether(
+            Number(razor_amount)
+          );
+        }
+      }
+    );
+
+  // for stakers
+    result.stakers.forEach(
+      async (Staker: {
+        staker: string | number;
+        stake: string;
+        sAmount: string | number;
+        totalSupply: string;
+      }) => {
+        const razor_amount = sRZR_to_RZR(
+          BigNumber.from(Staker.sAmount),
+          BigNumber.from(Staker.totalSupply),
+          BigNumber.from(Staker.stake)
+        );
+        //score will be based on the current stake in the block Number
+        if (!score[Staker.staker]) {
+          //if score[delegator] has no score setup already we will put it as intial amount
+          score[Staker.staker] = wei_to_ether(Number(razor_amount));
```

Comment:
The addresses from the subgraph are lowecased,  we need to return checksum addresses, you can use getAdress for that, example
https://github.com/snapshot-labs/snapshot-strategies/blob/master/src/strategies/synthetix-quadratic/index.ts#L124